bagger/src/bag_assembly.py

```diff
@@ -23,12 +23,11 @@ def cleanup(bag_details):
     if os.path.isdir(directory):
         shutil.rmtree(directory)
 
-    zipfile = bag_details["zip_file_path"]
-
-    logging.debug("deleting %s", zipfile)
-
-    if os.path.isfile(zipfile):
-        os.remove(zipfile)
+    zipfile = bag_details.get("zip_file_path", None)
+    if zipfile is not None:
+        logging.debug("deleting %s", zipfile)
+        if os.path.isfile(zipfile):
+            os.remove(zipfile)
```

Comment:
Python convention is to try the thing and catch the exception, rather than all these if checks.

i.e.,

```python
try:
    zipfile = bag_details["zip_file_path"]
    logging.debug("deleting %s", zipfile)
    try:
        os.remove(zipfile)
    except FileDoesNotExistError:
        pass
except KeyError:
    pass
```

But I don't think it makes a massive difference here.